subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarcPlugin.java

```diff
@@ -85,6 +86,12 @@ protected void configureTaskDefaults(CodeNarc task, String baseName) {
         configureReportsConventionMapping(task, baseName);
     }
 
+    @Override
+    protected void beforeApply() {
+        // Necessary to disambiguate the published variants of newer codenarc versions (including the default version)
+        project.getPluginManager().apply(JvmEcosystemPlugin.class);
```

Comment:
Should all code quality plugins just apply the JvmEcosystemPlugin since they're all expecting JVM-ecosystem rules?

Reply:
Makes sense. As long as they're resolving JVM dependencies to retrieve the tool jars, they should be applying this plugin.

Reply:
I'd be fine keeping this to only the codenarc plugin in order to avoid the changes in the RC.